lib/dashing/app.rb

```diff
@@ -72,10 +74,15 @@ def authenticated?(token)
 get '/events', provides: 'text/event-stream' do
   protected!
   response.headers['X-Accel-Buffering'] = 'no' # Disable buffering for nginx
-  stream :keep_open do |out|
-    settings.connections << out
+  connection_id = SecureRandom.uuid
+  settings.connection_events[connection_id] = []
+  stream do |out|
     out << latest_events
-    out.callback { settings.connections.delete(out) }
+    loop do
+      while event = settings.connection_events[connection_id].shift do
+        out << event unless out.closed?
+      end
+    end
```

Comment:
This loop is killed by Puma when connection is terminated -- in fact the method doesn't even complete. The `GET` request to `/events` is still logged by Sinatra, so on some level the application is aware of the connection being closed -- it would be great if the `connection_id` was removed from the set of `connnection_events` at this time but I couldn't figure out how to do this.